plugins/Kaleidoscope-MouseKeys/src/kaleidoscope/plugin/MouseKeysConfig.cpp

```diff
@@ -0,0 +1,143 @@
+  if (checker == 0xffffffff) {
+    Runtime.storage().put(settings_addr_, ::MouseKeys.settings_);
+    Runtime.storage().commit();
+  }
+
+  Runtime.storage().get(settings_addr_, ::MouseKeys.settings_);
+  return EventHandlerResult::OK;
+}
+
+// -----------------------------------------------------------------------------
+EventHandlerResult MouseKeysConfig::onFocusEvent(const char *command) {
+  // If the focus command is a request for help, provide the list of valid
+  // commands.
+  if (::Focus.handleHelp(command, PSTR("mousekeys.scroll_interval\n"
+                                       "mousekeys.init_speed\n"
+                                       "mousekeys.base_speed\n"
+                                       "mousekeys.accel_duration")))
+    return EventHandlerResult::OK;
+
+  // The length of the string `mousekeys.`:
+  constexpr uint8_t base_cmd_len = 10;
+
+  // If this is not a MouseKeys command, do nothing.
+  if (strncmp_P(command, PSTR("mousekeys."), base_cmd_len) != 0)
+    return EventHandlerResult::OK;
+  // Otherwise, advance the pointer to the subcommand.
+  command += base_cmd_len;
+
+  enum Command : uint8_t {
+    SCROLL_INTERVAL,
+    INIT_SPEED,
+    BASE_SPEED,
+    ACCEL_DURATION,
+  };
+  Command cmd;
+
+  // Parse the (sub)command.  If it's not a valid command, abort.
+  if (strcmp_P(command, PSTR("scroll_interval")) == 0)
+    cmd = Command::SCROLL_INTERVAL;
+  else if (strcmp_P(command, PSTR("init_speed")) == 0)
+    cmd = Command::INIT_SPEED;
+  else if (strcmp_P(command, PSTR("base_speed")) == 0)
+    cmd = Command::BASE_SPEED;
+  else if (strcmp_P(command, PSTR("accel_duration")) == 0)
+    cmd = Command::ACCEL_DURATION;
+  else
+    return EventHandlerResult::ABORT;
+
+  if (::Focus.isEOL()) {
+    // If there is no argument given, we send back the current value of the
+    // setting that was requested.
+    uint16_t val;
+    switch (cmd) {
+    case Command::SCROLL_INTERVAL:
+      val = ::MouseKeys.getScrollInterval();
+      break;
+    case Command::INIT_SPEED:
+      val = ::MouseKeys.getCursorInitSpeed();
+      break;
+    case Command::BASE_SPEED:
+      val = ::MouseKeys.getCursorBaseSpeed();
+      break;
+    case Command::ACCEL_DURATION:
+      val = ::MouseKeys.getCursorAccelDuration();
+      break;
+    default:
+      return EventHandlerResult::ABORT;
+    }
+    ::Focus.send(val);
+    return EventHandlerResult::EVENT_CONSUMED;
+  } else {
+    // If there is an argument, we read it, then pass it to the corresponding
+    // setter method of MouseKeys.
+    uint16_t arg;
+    ::Focus.read(arg);
+
+    switch (cmd) {
+    case Command::SCROLL_INTERVAL:
+      ::MouseKeys.setScrollInterval(arg);
+      break;
+    case Command::INIT_SPEED:
+      ::MouseKeys.setCursorInitSpeed(arg);
+      break;
+    case Command::BASE_SPEED:
+      ::MouseKeys.setCursorBaseSpeed(arg);
+      break;
+    case Command::ACCEL_DURATION:
+      ::MouseKeys.setCursorAccelDuration(arg);
+      break;
+    }
+  }
```

Comment:
I'm liking this structure of the focus event handler. The top-level branching being based on whether we get or set makes the code clearer than branching on `cmd` on the top, and get/set below.

Also loving the getter/setter methods to make the code even easier to read.

At some point, we should convert other similar plugins to this structure, this sets a very good example. Thank you!

Reply:
I tried a different branching structure that was even easier to read, in my opinion, but the cost for each Focus method call was so big that I settled on this one.  It saves a lot of PROGMEM.